### PR TITLE
feat(full-text-index): add analyzer and testcase of text index

### PR DESCRIPTION
**Pull request by ljcis**
### What problem does this PR solve?

Issue Number: reslove #141 

### What is changed and how it works?
It works as a dictionary generate by learning text content
### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B
- [x] Test cases to be added
- [ ] No code

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: engine/index/clv/analyzer.go [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+		a, err = getAnalyzer(dicPath, name, field, version)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		a = newAnalyzer(dicPath, name, field)
+		a.needSample = true
+		a.writeTree = NewTrieWriteTree()
+	}
+	a.version = version
```
Comment (lijieac) [inline engine/index/clv/analyzer.go:552]: “a.version = version”应该放置在上面的else语句中。同时需要保存到cache中（clvAnalyzerCache和clvAnalyzerVersion都需要保存相应信息），以便后面来的shard可以取到同一个analyzer。

File: engine/index/clv/analyzer.go [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+
+	b := dictPrefix(latestVersion + 1)
+
+	readFromDicTree(root, b, &items)
+	tb, err := mergeset.OpenTable(path.Join(a.path, a.measurement, a.field), nil, nil)
+	if err != nil {
+		return 0, err
+	}
+	defer tb.MustClose()
+	err = tb.AddItems(items)
```
Comment (lijieac) [inline engine/index/clv/analyzer.go:342]: This scenario needs to be considered:  tb.AddItems has not finished, the system failed.

File: engine/index/clv/analyzer.go [hunk 3]
```diff
@@ -0,0 +1,555 @@
+/*
+Copyright 2022 Huawei Cloud Computing Technologies Co., Ltd.
```
[the remark this answers is not in the document]
Comment (ljcis) [reply engine/index/clv/analyzer.go:2]: in  make licence-check , "COPYRIGHT_HEADER  := "Copyright 2022 Huawei Cloud Computing Technologies Co., Ltd."  
 2023 can't pass licence-check

File: engine/index/clv/analyzer.go (hunk 1 above)
Comment (ljcis) [reply engine/index/clv/analyzer.go]: 已修改

File: engine/index/clv/analyzer.go (hunk 2 above)
Comment (ljcis) [reply engine/index/clv/analyzer.go:338]: 已修改

**Review by lijieac — APPROVED** [review]

**Review by lkx-del — APPROVED** [review]

File: engine/index/clv/analyzer.go [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+	"github.com/pkg/errors"
+)
+
+const (
+	Blank   = 0
+	Default = 1
+	Qmax    = 7
+	T       = 100
+	N       = 500000
+	qmin    = 1
```
[the remark this answers is not in the document]
Comment (ljcis) [reply engine/index/clv/analyzer.go]: 已处理

**Pull request merged**